src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -503,11 +508,20 @@ sharded_store::get_subject_version_written_at(subject sub, schema_version ver) {
 
 ss::future<bool> sharded_store::delete_subject_version(
   subject sub, schema_version ver, force force) {
-    auto sub_shard{shard_for(sub)};
-    co_return co_await _store.invoke_on(
-      sub_shard, _smp_opts, [sub{std::move(sub)}, ver, force](store& s) {
+    auto schema = co_await get_subject_schema(sub, ver, include_deleted::yes);
+
+    auto result = co_await _store.invoke_on(
+      shard_for(sub), _smp_opts, [sub{std::move(sub)}, ver, force](store& s) {
           return s.delete_subject_version(sub, ver, force).value();
       });
+
+    auto remaining_subjects = co_await get_schema_subjects(
+      schema.id, include_deleted::yes);
```

Comment:
nitpick: `subject_versions_has_any_of` would be a cheaper call if it accepted an `include_deleted` arg.